### PR TITLE
Incluído página Salvando Dados

### PR DESCRIPTION
**Pull request by rochamarcelo**
Tradução completa da página Salvando Dados

### CODE REVIEW

**Comment by rochamarcelo** [comment]
Tradução para português

File: pt/orm/saving-data.rst [hunk 1]
```diff
@@ -1,28 +1,1228 @@
...
+.. php:namespace:: Cake\ORM
+
+.. php:class:: Table
+    :noindex:
+
+Depois que você :doc:`carregou seus dados</orm/retrieving-data-and-resultsets>`
+provavelmente vai querer atualizar e salvar as alterações.
+
+Visão Geral Sobre Salvando Dados
+=========================
```
Comment (markstory) [inline pt/orm/saving-data.rst]: Title underlines need to be the same length as the text above them.

File: pt/orm/saving-data.rst [hunk 2]
```diff
@@ -1,28 +1,1228 @@
...
+
+Aplicações geralmente terá algumas maneiras de como os dados são salvos. A
+primeira é, obviamente, atravéz de formulários web e a outra é por geração direta
+ou alterando dados no código para enviar ao banco de dados.
+
+Inserindo Dados
+--------------
```
Comment (markstory) [inline pt/orm/saving-data.rst]: Title underlines need to be the same length as the text above them.

File: pt/orm/saving-data.rst [hunk 3]
```diff
@@ -1,28 +1,1228 @@
...
+
+    $article->title = 'A New Article';
+    $article->body = 'This is the body of the article';
+
+    if ($articlesTable->save($article)) {
+        // The $article entity contains the id now
+        $id = $article->id;
+    }
+Atualizando Dados
+-------------
```
Comment (markstory) [inline pt/orm/saving-data.rst]: Title underlines need to be the same length as the text above them.

File: pt/orm/saving-data.rst [hunk 4]
```diff
@@ -1,28 +1,1228 @@
...
+    $tag2->name = 'awesome';
+
+    $article = $articlesTable->get(12);
+    $article->comments = [$firstComment, $secondComment];
+    $article->tags = [$tag1, $tag2];
+
+    $articlesTable->save($article);
+
+Associe Muitos para Muitos (N para N) registros
+------------------------------
```
Comment (markstory) [inline pt/orm/saving-data.rst]: Title underlines need to be the same length as the text above them.

File: pt/orm/saving-data.rst [hunk 5]
```diff
@@ -1,28 +1,1228 @@
...
+na associação::
+
+    $tag1 = $articlesTable->Tags->findByName('cakephp')->first();
+    $tag2 = $articlesTable->Tags->newEntity();
+    $tag2->name = 'awesome';
+
+    $articlesTable->Tags->link($article, [$tag1, $tag2]);
+ 
+Salvando Dados da Tabela de Ligação
+-----------------------------
```
Comment (markstory) [inline pt/orm/saving-data.rst]: Title underlines need to be the same length as the text above them.

File: pt/orm/saving-data.rst [hunk 6]
```diff
@@ -1,28 +1,1228 @@
...
+                '_joinData' => [
+                    'tagComment' => 'Great article!',
+                ]
+            ],
+        ]
+    ];
+    $articlesTable->newEntity($data, ['associated' => ['Tags']]);
+
+Remover Associação Muitos para Muitos (N para N) Registros
+---------------------------
```
Comment (markstory) [inline pt/orm/saving-data.rst]: Title underlines need to be the same length as the text above them.

File: pt/orm/saving-data.rst [hunk 7]
```diff
@@ -1,28 +1,1228 @@
...
+
+Quando modificando registros, configurando ou alterando diretamente as propriedades,
+nenhuma validação é realizada, que é um problema quando está aceitando dados de 
+formulário. As seções seguintes demostrarão como converter eficientemente dados de 
+formulário em entidades que podem ser validadas e salva.
+
+.. _converting-request-data:
+
+Convertendo Dados de Requisição em Entidades
+=====================================
```
Comment (markstory) [inline pt/orm/saving-data.rst]: Title underlines need to be the same length as the text above them.

File: pt/orm/saving-data.rst [hunk 8]
```diff
@@ -1,28 +1,1228 @@
...
 
-.. figure:: /_static/img/save-cycle.png
-   :align: left
-   :alt: Flow diagram showing the save process.
+Você sempre pode contar de obter uma entidade de volta com ``newEntity()``. Se a validação
+falhar, sua entidade conterá erros, e quaisquer campos inválidos não serão preenchidos 
+na entidade criada.
+
+Convertendo Dados de Associação BelongsToMany
+-----------------------------
```
Comment (markstory) [inline pt/orm/saving-data.rst]: Title underlines need to be the same length as the text above them.

File: pt/orm/saving-data.rst [hunk 9]
```diff
@@ -1,28 +1,1228 @@
...
+serão objetos novos, e as outras duas serão referências a registros existentes.
+
+Ao converter dados belongsToMany, você pode desativar a criação de nova entidade, usando
+a opção ``onlyIds``. Quando habilitado, esta opção restringe transformação de
+belongsToMany para apenas usar a chave ``_ids`` e ignorar todos os outros dados.
+
+.. versionadded:: 3.1.0
+    A opção ``onlyIds`` foi adicionada na versão 3.1.0
+
+Convertendo Dados de Associação HasMany
```
Comment (markstory) [inline pt/orm/saving-data.rst]: Title underlines need to be the same length as the text above them.

Comment (rochamarcelo) [inline pt/orm/saving-data.rst]: Sorry @markstory, my last commit should fix it.

Comment (markstory) [inline pt/orm/saving-data.rst]: No worries. It is a very common mistake we all make :smile:

**Comment by markstory** [comment]
There is another build error

```
/data/pt/orm/saving-data.rst:795: ERROR: Unexpected indentation.
```

File: pt/orm/saving-data.rst (hunk 9 above)
Comment (rochamarcelo) [inline pt/orm/saving-data.rst]: @markstory  could you help me with the error: "ERROR: Unexpected indentation.". I don't know how to fix it.

Comment (rochamarcelo) [inline pt/orm/saving-data.rst]: It looks the same as in the english version

Comment (rochamarcelo) [inline pt/orm/saving-data.rst]: @markstory, I fixed the error "ERROR: Unexpected indentation." but now I got the error: "/data/pt/orm/saving-data.rst:182: WARNING: undefined label: entities-mass-assignment (if the link has no caption the label must precede a section header)"

**Comment by markstory** [comment]
Thanks @rochamarcelo !

**Pull request merged**